src/walletConnect/screens/ActionRequest.tsx

```diff
@@ -83,7 +83,7 @@ function ActionRequest({ navigation, route: { params: routeParams } }: Props) {
       : method === SupportedActions.personal_decrypt
       ? Buffer.from(params[1]).toString('hex')
       : method === SupportedActions.personal_sign
-      ? params[0]
+      ? Buffer.from(params[0].replace(/^0x/, ''), 'hex').toString() || t('action.invalidMessage')
```

Comment:
I propose we display the raw param in case hex decoding fails, instead of "Invalid message", which is not entirely correct.
Just in case the dapp wants to sign something that's not a string. Not sure how likely it is to happen but at least we try to display it.
And the last fallback would be "Empty message" if still `null | undefined | ''`

```suggestion
      ? Buffer.from(trimLeading0x(params[0]), 'hex').toString() || params[0] || t('action.emptyMessage')
```

We also have a `trimLeading0x` util.